### PR TITLE
Edited User Guide's "Modify" section 

### LINKED ISSUE #108: Update User Guide Yong Ning
Please make sure that your relevant section fulfills these rubric requirements:
- User-centric description of feature/function/command   
- Easy-to-follow step-by-step instructions with clear headings   
- High consideration of user needs by providing other relevant and helpful information (e.g. warnings, scenarios explaining what to do if a command is invalid or fails, tips, important note)   
- Relevant, and helpful graphics; visual and/or textual cues are used to indicate which graphic(s) go with which set of instructions 
- ANNOTATED screenshots for certain commands (delete don't need)
- Add shortcuts back to table of contents 
- Examples that are effective, customized to the application and notes are provided regarding warnings and tips
- YOU LANGUAGE + WHY/CONTEXT of usage and not just HOW ("You can update details of an existing internship entry to keep track of changes" vs "Update details of an existing internship entry")

### LINKED ISSUE #146: [PE-D][Tester C] modify requires at least 1 field to be specified
**Steps to reproduce:**
In the UG under `modify`, the constraints state under the 3rd bullet point that "Not all categories need to be specified." This implies that it should be acceptable if none of the categories are specified. However, when attempting to modify an entry without any category specified, an error that was not included in the UG constraints is thrown 

**Expected behavior:**

Based on the UG description, should be allowed to modify with no category specified.

**Actual behavior:**

Not allowed to modify with no category specified.

Perhaps you can update the UG to correctly reflect this behavior.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: craigtonlian/ped#3

### LINKED ISSUE #182: [PE-D][Tester A] Command format is put in the example section in command summary
In the command summary in UG, the Format, Examples section of Modify puts the format in both format and example section.

A good way to improve is to change the command after e.g. into an example.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: ChangruHenryQian/ped#8

### LINKED ISSUE #186: [PE-D][Tester D] Unexpected Output for Modify Command
When modifying an internship application with the same data for certain fields (even with different data for duration), the system responds with the confusing message "This internship application already exists in the tracker." The expected output should be no response, as it was not stated in the User Guide that the system will notify the user if the modify command actually did nothing.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: joeng03/ped#3

### PR DESCRIPTION
**Pull request by yongning0310**
The previous user guide for the 'modify' section was inconsistent with the rest of the user guide, making it more challenging for users to read.

Ensuring consistency in the styling of the user guide is important for ease of reading.

I have edited the 'modify' section to align it with the rest of the user guide.

@jinyang628  can consider moving the explanation of what constitutes a duplicate entry to the 'parameter requirements' section, as the same table is needed for the explanation in the 'modify' section.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -266,21 +266,37 @@ so that you can track your past internship applications for your own reference.
...
 <br>
 
+
 ### Editing an Internship: `modify`
 
-Update details of an existing internship entry.
+Need to update an existing internship entry? The `modify` command lets you adjust the details of your applications in Flagship. Whether the application deadline has changed or you've picked up a new skill that meets the job requirements, ensure your internship details are current.
 
-**Format**: `modify INDEX c/COMPANY_NAME ro/ROLE a/APPLICATION_STATUS s/START_DATE de/DEADLINE du/DURATION re/REQUIREMENTS...`
+**Format**: `modify INDEX c/COMPANY_NAME ro/ROLE a/APPLICATION_STATUS de/DEADLINE s/START_DATE du/DURATION [re/REQUIREMENTS]...`
```
Comment (jinyang628) [inline docs/UserGuide.md:276]: Thanks for rectifying the order

File: docs/UserGuide.md [hunk 2]
```diff
@@ -266,21 +266,37 @@ so that you can track your past internship applications for your own reference.
...
+⚠️ Flagship does not permit editing the company name and role in a way that creates a duplicate of an existing internship entry with the same company and role. This policy ensures consistency with the "create" section, where the creation of duplicate internship entries is also prohibited.
+</div>
+
+<div markdown="span" class="alert alert-warning">
+⚠️ There must be at least one parameter following the INDEX. If no parameters are provided after the INDEX, a warning will be issued.
+</div>
+
+<div markdown="block" class="alert alert-info">
+ℹ️ Utilize the `modify` command to keep your internship applications up-to-date. It's an essential part of managing your internship timeline effectively in Flagship.
+</div>
```
Comment (jinyang628) [inline docs/UserGuide.md:292]: Not the best usage of this alert-info block. This sort of use case related information please include it directly in the introduction of ur command

File: docs/UserGuide.md [hunk 3]
```diff
@@ -266,21 +266,37 @@ so that you can track your past internship applications for your own reference.
...
+
+<div markdown="span" class="alert alert-warning">
+⚠️ Flagship does not permit editing the company name and role in a way that creates a duplicate of an existing internship entry with the same company and role. This policy ensures consistency with the "create" section, where the creation of duplicate internship entries is also prohibited.
+</div>
+
+<div markdown="span" class="alert alert-warning">
+⚠️ There must be at least one parameter following the INDEX. If no parameters are provided after the INDEX, a warning will be issued.
+</div>
+
```
Comment (jinyang628) [inline docs/UserGuide.md]: Let's refrain from putting these information in separate blocks. Just put it under one and list it neatly. Up to u how u want to present it (bullet point style or sth else)

**Review by jinyang628 — CHANGES_REQUESTED** [review]
1. Missing ANNOTATED screenshot (look at how javin does his)
2. Could u make this change u mentioned in ur PR directly and i will see how it looks on the website post-merge ("moving the explanation of what constitutes a duplicate entry to the 'parameter requirements' section, as the same table is needed for the explanation in the 'modify' section")

File: docs/UserGuide.md (hunk 3 above)
Comment (yongning0310) [reply docs/UserGuide.md]: included all three in bullet point style, under the same alert-warning

File: docs/UserGuide.md (hunk 2 above)
Comment (yongning0310) [reply docs/UserGuide.md:292]: noted. removed this alert-info block

File: docs/UserGuide.md (hunk 1 above)
Comment (yongning0310) [reply docs/UserGuide.md:276]: np

File: docs/UserGuide.md [hunk 4]
```diff
@@ -259,10 +281,6 @@ our definition of identical internships further, Flagship will not be able to ca
 as effectively.
 </div>
 
-<div markdown="block" class="alert alert-info">
-ℹ️ We allow you to create internship entries with application deadlines and start dates before today's current date. It is designed as such 
-so that you can track your past internship applications for your own reference. 
-</div>
```
Comment (jinyang628) [inline docs/UserGuide.md]: Is there a reason you deleted this? I inserted this specifically to guard against PE attacks

**Review by jinyang628 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md [hunk 5]
```diff
@@ -259,28 +281,35 @@ our definition of identical internships further, Flagship will not be able to ca
...
+
+⚠️ **Internship Entry Modification Policies:**
+
+- **INDEX is Mandatory**: The INDEX, a positive integer, is required to identify the entry's position in the list. Modifying an entry will replace the existing list, not add to it.
+- **No Duplicate Company Name and Role**: Editing to create a duplicate entry with the same company name and role as an existing internship entry is not permitted, ensuring consistency with the creation section's prohibition of duplicates.
+- **Parameter Requirement**: There must be at least one parameter following the INDEX when modifying an entry. Omission of parameters will trigger a warning.
+</div>
+
+**Expected Output**:
+![Sort](images/Modify.png)
```
Comment (jinyang628) [inline docs/UserGuide.md:296]: I dont think u labelled this correctly. It should be modify

**Review by jinyang628 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 5 above)
Comment (yongning0310) [reply docs/UserGuide.md:296]: corrected to modify

File: docs/UserGuide.md (hunk 4 above)
Comment (yongning0310) [reply docs/UserGuide.md]: added back

**Review by jinyang628 — APPROVED** [review]
LGTM

**Pull request merged**